### PR TITLE
Modifying group selection component

### PR DESCRIPTION
**Pull request by n-germain-work**
image

### CODE REVIEW

File: libs/ui/src/lib/static-autocomplete/group/detailed/detailed.component.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+export class DetailedGroupComponent implements OnInit {
+  groups$ = new BehaviorSubject<StaticGroup[]>([]);
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public data: { items: string[], scope: GroupScope },
+    public dialogRef: MatDialogRef<DetailedGroupComponent>
+  ) { }
+
+  ngOnInit() {
+    const groups = JSON.parse(JSON.stringify(staticGroups[this.data.scope] ?? {}));
```
Comment (phpgeekfr) [inline libs/ui/src/lib/static-autocomplete/group/detailed/detailed.component.ts:21]: do you need JSON.parse & stringify ?

Maybe just cloning is enough ?
```
const groups = { ...staticGroups[this.data.scope] };
```

File: libs/ui/src/lib/static-autocomplete/group/detailed/detailed.component.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        if (!fullGroup) {
+          group.items = group.items.filter(item => this.data.items.includes(item));
+        }
+      }
+      this.groups$.next(groups);
+    }
+  }
+
+  close() {
+    this.dialogRef.close()
```
Comment (phpgeekfr) [inline libs/ui/src/lib/static-autocomplete/group/detailed/detailed.component.ts:35]:
```suggestion
    this.dialogRef.close();
```

File: libs/ui/src/lib/static-autocomplete/group/group.component.ts [hunk 3]
```diff
@@ -1,290 +1,200 @@
...
+    const pastedText = clipboardData.getData('text');
+    const pastedValues = pastedText.split(',').map(item => item.trim());
+    const pastedItems = new Set(
+      pastedValues
+        .map(item => {
+          for (const group of this.groups) {
+            if (group.label.toLowerCase() === item.toLowerCase()) return group.items;
+            if (group.items.includes(item.toLowerCase())) return item.toLowerCase();
+          }
+        })
```
Comment (phpgeekfr) [inline libs/ui/src/lib/static-autocomplete/group/group.component.ts:182]:
```suggestion
        .map(item => item.toLowerCase())
        .map(item => {
          for (const group of this.groups) {
            if (group.label.toLowerCase() === item) return group.items;
            if (group.items.includes(item)) return item;
          }
        })
```

File: libs/ui/src/lib/static-autocomplete/group/detailed/detailed.component.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import { ChangeDetectionStrategy, Component, Inject, OnInit } from '@angular/core';
```
Comment (phpgeekfr) [inline libs/ui/src/lib/static-autocomplete/group/detailed/detailed.component.ts:1]: this component is a duplicate of `libs/contract/src/lib/term/components/detailed` 

you should move it to `libs/ui/src/lib` and use the same

File: libs/contract/src/lib/avails/form/avails.form.ts [hunk 5]
```diff
@@ -46,7 +46,7 @@ function createTerritoriesControl(territories: Territory[] = []) {
 
 function createBaseAvailControl(avail: Partial<BaseAvailsFilter> = {}) {
   return {
-    medias: new FormStaticValueArray<'medias'>(avail.medias, 'medias', [Validators.required, Validators.minLength(0)]),
+    medias: new FormStaticValueArray<'medias'>(avail.medias || [], 'medias', [Validators.required, Validators.minLength(0)]),
```
[the remark this answers is not in the document]
Comment (n-germain-work) [reply libs/contract/src/lib/avails/form/avails.form.ts:49]: I checked below forms and manage to find my test avails :
image

File: libs/contract/src/lib/contract/holdback/selection-modal/selection-modal.component.ts [hunk 6]
```diff
@@ -35,7 +36,7 @@ export class SelectionModalComponent implements OnInit {
   ) {}
 
   ngOnInit() {
-    this.form = FormList.factory(this.data.holdbacks, (holdback) => new HoldbackForm(holdback), []);
+    this.form = FormList.factory(this.data.holdbacks, (holdback) => new HoldbackForm(holdback), Validators.required);
```
[the remark this answers is not in the document]
Comment (n-germain-work) [reply libs/contract/src/lib/contract/holdback/selection-modal/selection-modal.component.ts:39]: I'm sending the `required` parameter to my new component to display the linked error.

libs\contract\src\lib\contract\holdback\selection-modal\selection-modal.component.html
image

libs\ui\src\lib\static-autocomplete\group\group.component.html
image

File: libs/movie/src/lib/movie/components/request-asking-price/request-asking-price.component.html [hunk 7]
```diff
@@ -4,33 +4,23 @@ <h2>Request Asking Price</h2>
...
-      <mat-icon matPrefix svgIcon="world"></mat-icon>
-      <static-group
-        [formControl]="form.get('territories')"
-        scope="territories"
-        displayAll="World"
-        placeholder="Select Territories"
-        required>
-      </static-group>
-      <mat-hint>Select the territories you are interested in.</mat-hint>
-      <mat-error *ngIf="form.get('territories').hasError('required')">Select at least one territory</mat-error>
```
[the remark this answers is not in the document]
Comment (n-germain-work) [reply libs/movie/src/lib/movie/components/request-asking-price/request-asking-price.component.html:18]: I forgot the `mat-hint`... ^^
The `mat-error` text has been replaced as below :
image

Gonna check with design and adapt if needed 👍

[the remark this answers is not in the document]
Comment (phpgeekfr) [reply libs/movie/src/lib/movie/components/request-asking-price/request-asking-price.component.html:18]: the sentence can maybe be put in the label or placeholder instead ?

[the remark this answers is not in the document]
Comment (n-germain-work) [reply libs/movie/src/lib/movie/components/request-asking-price/request-asking-price.component.html:18]: adapting ^^

File: libs/ui/src/lib/static-autocomplete/group/detailed/detailed.module.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import { MatButtonModule } from '@angular/material/button';
+import { MatIconModule } from '@angular/material/icon';
+import { MatChipsModule } from '@angular/material/chips';
+import { MatDialogModule } from '@angular/material/dialog';
+
+@NgModule({
+  imports: [
+    CommonModule,
+    ToLabelModule,
+    FlexLayoutModule,
```
[the remark this answers is not in the document]
Comment (n-germain-work) [reply libs/ui/src/lib/static-autocomplete/group/detailed/detailed.module.ts]: `CommonModule` seems needed for the `titlecase ` pipe.
Same for `ToLabelModule` and `toLabel` pipe.

File: libs/ui/src/lib/static-autocomplete/group/detailed/detailed.module.ts [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    GlobalModalModule,
+    MatButtonModule,
+    MatIconModule,
```
[the remark this answers is not in the document]
Comment (n-germain-work) [reply libs/ui/src/lib/static-autocomplete/group/detailed/detailed.module.ts]: now used by the 'paste' button.

File: libs/ui/src/lib/static-autocomplete/group/detailed/detailed.module.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (phpgeekfr) [reply libs/ui/src/lib/static-autocomplete/group/detailed/detailed.module.ts]: Sry, I was talking about FlexLayoutModule

File: apps/catalog/catalog-e2e/src/integration/marketplace/deal-display-offer.cy.ts [hunk 10]
```diff
@@ -58,7 +58,9 @@ describe('Deal negociation', () => {
   it('Buyer and Seller can acces the offer / sale page', () => {
     browserAuth.signinWithEmailAndPassword(buyer.user.email);
     cy.visit('');
-    get('skip-preferences').click();
+    firestore.get('users/0-e2e-buyerOrgAdminUid').then((user: User) => {
```
Comment (phpgeekfr) [inline apps/catalog/catalog-e2e/src/integration/marketplace/deal-display-offer.cy.ts:61]:
```suggestion
    **firestore.get(`users/${buyer.user.uid}`).then((user: User) => {
```

File: libs/ui/src/lib/static-autocomplete/group/group.component.ts [hunk 11]
```diff
@@ -1,290 +1,205 @@
...
-    const validators = this._required ? [Validators.required] : []
-    this.form.setValidators(validators)
-    this.stateChanges.next();
-  }
+  @Input() label: string;
+  @Input() selectIcon = 'world';
+  @Input() filterPlaceholder: string;
+  @Input() displayAll: string;
+  @Input() required = false;
+  @Input() requiredMsg = ' This field is mandatory ';
```
Comment (phpgeekfr) [inline libs/ui/src/lib/static-autocomplete/group/group.component.ts:26]: why not add the spaces in html directly ? or in css

**Review by phpgeekfr — APPROVED** [review]
can't you get rid of  `DetailedTermsModule` ?

**Pull request merged**